depmap-compute/depmap_compute/slice.py

```diff
@@ -38,3 +38,26 @@ def decode_slice_id(slice_id) -> tuple[str, str, str]:
     slice_type = unquote(parts[3])  # "label", "entity_id", or "transpose_label"
 
     return dataset_id, dimension_identifier, slice_type
+
+
+def slice_id_to_slice_query(slice_id: str) -> SliceQuery:
+    """Take a legacy slice ID string and convert it to the newer slice query format."""
+    dataset_id, dimension_identifier, slice_type = decode_slice_id(slice_id)
+
+    # Slice query identifier types use different (more descriptive) terminology
+    if slice_type == "label":
+        slice_query_identifier_type = "feature_label"
+    elif slice_type == "transpose_label":
+        # "transpose_label" identifiers are always depmap IDs
+        # In breadbox, this is stored as an ID, not a label.
+        slice_query_identifier_type = "sample_id"
+    elif slice_type == "entity_id":
+        # "entity_id" is only used in older parts of the codebase (not DE2 or ContextManager)
+        assert not dataset_id.startswith("breadbox/"), "Breadbox datasets do not support lookups by entity_id"
```

Comment:
Unrelated fix: I noticed that this function had gotten deleted somehow, even though it's imported by the `portal-backend`, which was only working because it was using an older version of this `depmap-compute `package that had the function. 

I also updated this function to ensure that breadbox datasets are never queried with the `"entity_id"` slice type (see my above comment for more background on what that's a potential issue). I _believe_ the only parts of our codebase that use the `entity_id` slice type at this point are:
* Genomic Associations 
* Constellation